File: android/app/src/main/java/com/magicforms/mobile/data/models/SignatureModels.kt
package com.magicforms.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SignaturesListResponse(
    val ok: Boolean,
    val limit: Int = 200,
    val count: Int = 0,
    val signatures: List<UserSignatureItem> = emptyList(),
)

@Serializable
data class UserSignatureItem(
    val id: Int,
    val label: String = "",
    @SerialName("image_url") val imageUrl: String = "",
    @SerialName("is_primary") val isPrimary: Boolean = false,
    @SerialName("sort_order") val sortOrder: Int = 0,
    @SerialName("created_at") val createdAt: String = "",
    @SerialName("created_at_label") val createdAtLabel: String = "",
)

@Serializable
data class SignatureMutationResponse(
    val ok: Boolean,
    val message: String? = null,
    val signature: UserSignatureItem? = null,
)

File: ios/MagicFormsMobile/Models/SignatureModels.swift
import Foundation

struct SignaturesListResponse: Decodable {
    let ok: Bool
    let limit: Int
    let count: Int
    let signatures: [UserSignatureItem]
}

struct UserSignatureItem: Decodable, Identifiable, Hashable {
    let id: Int
    let label: String
    let imageUrl: String
    let isPrimary: Bool
    let sortOrder: Int
    let createdAt: String
    let createdAtLabel: String

    enum CodingKeys: String, CodingKey {
        case id, label
        case imageUrl = "image_url"
        case isPrimary = "is_primary"
        case sortOrder = "sort_order"
        case createdAt = "created_at"
        case createdAtLabel = "created_at_label"
    }
}

struct SignatureMutationResponse: Decodable {
    let ok: Bool
    let message: String?
    let signature: UserSignatureItem?
}
